sky/clouds/aws.py

```diff
@@ -292,30 +290,34 @@ def check_credentials(self) -> Tuple[bool, Optional[str]]:
         except ImportError:
             raise ImportError('Fail to import dependencies for AWS.'
                               'Try pip install "skypilot[aws]"') from None
+        prefix = '    '
         help_str = (
             ' Run the following commands:'
-            '\n      $ pip install boto3'
-            '\n      $ aws configure'
-            '\n    For more info: '
+            f'\n{prefix}  $ pip install boto3'
+            f'\n{prefix}  $ aws configure'
+            f'\n{prefix}For more info: '
             'https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html'  # pylint: disable=line-too-long
         )
-        # This file is required because it will be synced to remote VMs for
-        # `aws` to access private storage buckets.
-        # `aws configure list` does not guarantee this file exists.
-        if not os.path.isfile(os.path.expanduser('~/.aws/credentials')):
-            return (False, '~/.aws/credentials does not exist.' + help_str)
 
         # Checks if the AWS CLI is installed properly
-        try:
-            _run_output('aws configure list')
-        except subprocess.CalledProcessError:
+        proc = subprocess.run('aws configure list',
+                              shell=True,
+                              check=False,
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE)
+        if proc.returncode != 0:
+            additional_help = f'\n{prefix}Credentials may also need to be set.' + help_str
+            if 'SSO' in proc.stderr.decode().split():
+                additional_help = (
+                    f'\n{prefix}You are using SSO account, please update the awscli package with:'
+                    f'\n{prefix}  $ pip install awscli>=1.27.10 boto3')
             return False, (
-                'AWS CLI is not installed properly.'
-                ' Run the following commands under sky folder:'
+                'AWS CLI is not installed properly. '
+                'Run the following commands under sky folder:'
                 # TODO(zhwu): after we publish sky to PyPI,
                 # change this to `pip install sky[aws]`
-                '\n     $ pip install .[aws]'
-                '\n   Credentials may also need to be set.' + help_str)
+                f'\n{prefix}  $ pip install .[aws]'
```

Comment:
That is quite tricky: do we want to have our master branch user see the correct message or the PyPI user? If we put `skypilot` here, the master branch user will not be able to correctly install it.

Reply:
I think we should make PyPI users see the correct message, since users installing from master can be assumed as more "hands-on"/advanced, so they should be able to change the command accordingly?